lib/logstash/outputs/s3.rb

```diff
@@ -180,6 +189,11 @@ def create_temporary_file
       end
 
       @tempfile = File.open(filename, "a")
+      
+      #  header row processing
+      if @header_row != "" 
+        @tempfile.syswrite(@header_row + "\n")
+      end
```

Comment:
I assume we only want this to be added once in the file correct?

We are opening the file in `append` mode, if something weird happen to logstash and it has to restart we could have multiple headers added to the same file. This problem is rare since the restart has to happen in a one minute window. (see how we generate the filename https://github.com/logstash-plugins/logstash-output-s3/pull/54/files#diff-40972c051aeee75a31b999f62e7bcd65L279)

Might be worth it to check if there is already an header written to the file before adding a new one?

Also would you mind adding a test for this new feature?